vmdk_plugin/drivers/vmdk/vmdkops/vmdkops.go

```diff
@@ -59,14 +60,32 @@ func (v VmdkOps) Remove(name string, opts map[string]string) error {
 	return err
 }
 
-// Attach a volume
-func (v VmdkOps) Attach(name string, opts map[string]string) ([]byte, error) {
+// RawAttach attaches a volume and returns a raw response.
+func (v VmdkOps) RawAttach(name string, opts map[string]string) ([]byte, error) {
 	log.Debugf("vmdkOps.Attach name=%s", name)
 	str, err := v.Cmd.Run("attach", name, opts)
 	if err != nil {
+		log.WithFields(log.Fields{"name": name, "opts": opts, "error": err}).Error("RawAttach failed ")
 		return nil, err
 	}
-	return str, err
+	return str, nil
```

Comment:
`Attach(..)` now returns a `VolumeDevSpec` by converting the `[]byte` response from `RawAttach(..)`. `RawAttach` is made public only for mocking purposes, where `string` device paths are used.